x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwkValidateUtil.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authc.jwt;
+
+import com.nimbusds.jose.Algorithm;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.crypto.MACSigner;
+import com.nimbusds.jose.jwk.Curve;
+import com.nimbusds.jose.jwk.ECKey;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.KeyOperation;
+import com.nimbusds.jose.jwk.KeyUse;
+import com.nimbusds.jose.jwk.OctetSequenceKey;
+import com.nimbusds.jose.jwk.RSAKey;
+import com.nimbusds.jose.jwk.gen.RSAKeyGenerator;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.SettingsException;
+import org.elasticsearch.core.Tuple;
+import org.elasticsearch.xpack.core.security.authc.jwt.JwtRealmSettings;
+
+import java.security.PublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Utilities for JWK Validation.
+ */
+public class JwkValidateUtil {
+    private static final Logger LOGGER = LogManager.getLogger(JwkValidateUtil.class);
+
+    // Static method for unit testing. No need to construct a complete RealmConfig with all settings.
+    static Tuple<List<String>, List<JWK>> filterJwksAndAlgorithms(final List<JWK> jwks, final List<String> algs) throws SettingsException {
```

Comment:
Since we are doing this only once per JWKS loading round, it might be useful to log why each key was filtered rejected so that the admin knows or can find out  ? No need to do _now_, but adding as a suggestion nevertheless